instructor/compiler.py

```diff
@@ -0,0 +1,469 @@
+        ret[k] = {'parser_ab': []}
+
+    cyto = []
+    nuclear = []
+    whole = []
+    for antibodies in a.western_blot.antibodies.all():
+        for ps in a.strain_protocol.filter(enabled=True):
+            bands = WesternBlotAntibodyBands.objects.filter(antibody=antibodies, strain_protocol=ps)
+            cyto_marks = []
+            wcl_marks = []
+            nuc_marks = []
+            for band in bands:
+                if band.is_background is False:
+                    cyto_marks.append({
+                        'weight': band.cyto_weight,
+                        'intensity': band.cyto_intensity,
+                        'primary_anti_body': ['AB_{}'.format(antibodies.id)]
+                    })
+                    wcl_marks.append({
+                        'weight': band.wcl_weight,
+                        'intensity': band.wcl_intensity,
+                        'primary_anti_body': ['AB_{}'.format(antibodies.id)]
+                    })
+                    nuc_marks.append({
+                        'weight': band.nuc_weight,
+                        'intensity': band.nuc_intensity,
+                        'primary_anti_body': ['AB_{}'.format(antibodies.id)]
+                    })
+            cyto.append({
+                'identifier': "SP_ID_{}".format(ps.id),
+                'marks': cyto_marks
+            })
+            nuclear.append({
+                'identifier': "SP_ID_{}".format(ps.id),
+                'marks': nuc_marks
+            })
+            whole.append({
+                'identifier': "SP_ID_{}".format(ps.id),
+                'marks': nuc_marks
+            })
+
+    ret['cyto']['parser_ab'] = cyto
+    ret['nuclear']['parser_ab'] = nuclear
+    ret['whole']['parser_ab'] = whole
+
+    return ret
+
+
+def gel_types(a):
+    ret = []
+    if a.western_blot.has_gel_10:
+        ret.append('.10')
+    if a.western_blot.has_gel_12:
+        ret.append('.12')
+    if a.western_blot.has_gel_15:
+        ret.append('.15')
+    return ret
+
+
+def micro_kinds(a):
+    ret = {}
+    for sp in a.microscopy_sample_prep.all():
+        analysis = sp.analysis
+        condition = sp.condition
+        if not ret.has_key(analysis):
+            ret[analysis] = {
+                'name': analysis,
+                'conditions': {
+
+                },
+                'identifiers': {}
+            }
+        ret[analysis]['conditions'][condition] = {
+            'name': condition,
+            'short_name': condition
+        }
+        if not ret[analysis]['conditions'][condition].has_key('identifiers'):
+            ret[analysis]['conditions'][condition]['identifiers'] = {}
+        identifiers = {}
+        for p in sp.microscopy_images.all():
+            ret[analysis]['identifiers']["SP_ID_{}".format(p.strain_protocol_id)] = 1
+            ret[analysis]['conditions'][condition]['identifiers']["SP_ID_{}".format(p.strain_protocol_id)] = 1
+    return ret
+
+
+def lysate_kinds(a):
```

Comment:
name should be assignment, or whatever it is actually suppose to represent, a doc string again, would help a lot